File: solutions/python/sieve/1/sieve.py
def primes(limit):
    prime = [True] * (limit + 1)
    i = 2
    while i * i <= limit:
        if prime[i]:
            for j in range(i * i, limit + 1, i):
                prime[j] = False
        i += 1
    result = []
    for j in range(2, limit + 1):
        if prime[j]:
            result.append(j)
    return result

File: solutions/cpp/sieve/1/sieve.cpp
#include "sieve.h"
#include<vector>
namespace sieve {

// TODO: add your solution here
    bool check_prime(int num){
        for(int i = 2; i < num / 2 + 1; i++){
            if (num % i == 0) {return false;}
        }
        return true;
    }
    std::vector<int> primes(int num){
        std::vector<int> result;
        for(int i = 2; i <= num; i++){
            if(check_prime(i)) {
                result.emplace_back(i);
            }
        }
        return result;
    }

}  // namespace sieve
